pcapgo/ngwrite_dsb.go

```diff
@@ -0,0 +1,113 @@
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ 0 |                   Block Type = 0x0000000A                     |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ 4 |                      Block Total Length                       |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ 8 |                          Secrets Type                         |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+12 |                         Secrets Length                        |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+16 /                                                               /
+   /                          Secrets Data                         /
+   /              (variable length, padded to 32 bits)             /
+   /                                                               /
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   /                                                               /
+   /                       Options (variable)                      /
+   /                                                               /
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   /                       Block Total Length                      /
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+	Block Type: The block type of the Decryption Secrets Block is 10.
+
+	Block Total Length: total size of this block, as described in {{section_block}}.
+
+	Secrets Type (32 bits): an unsigned integer identifier that describes the format of the following Secrets field. Requests for new Secrets Type codes should be made by creating a pull request to update this document as described in {{section_block_code_registry}}.
+
+	Secrets Length (32 bits): an unsigned integer that indicates the size of the following Secrets field, without any padding octets.
+
+	Secrets Data: binary data containing secrets, padded to a 32 bit boundary.
+
+	Options: optionally, a list of options (formatted according to the rules defined in {{section_opt}}) can be present. No DSB-specific options are currently defined.
+*/
+
+const (
+	PcapngBlockHeadersize            = 8 // block type + block total length
+	PcapngDecryptionSecretsBlockSize = 8 // Secrets type + Secrets length
+)
+
+// pcapngBlockHeader is the header of a pcapng block.
+type pcapngBlockHeader struct {
+	blockType        uint32
+	blockTotalLength uint32
+}
+
+// pcapngDecryptionSecretsBlock is the header of a section.
+type pcapngDecryptionSecretsBlock struct {
+	secretsType   uint32
+	secretsLength uint32
+}
+
+// WriteDecryptionSecretsBlock writes a Decryption Secrets Block to the writer.
+func (w *NgWriter) WriteDecryptionSecretsBlock(secretType uint32, secretPayload []byte) error {
+
+	switch secretType {
+	case DSB_SECRETS_TYPE_SSH, DSB_SECRETS_TYPE_ZIGBEE_NWK_KEY, DSB_SECRETS_TYPE_WIREGUARD, DSB_SECRETS_TYPE_ZIGBEE_APS_KEY:
```

Comment:
Is there any reason to forbid these other secret types?

Reply:
Unverified functionality yet.
I think that error message is better than unknow error info.

Reply:
I have verified that it also works for `DSB_SECRETS_TYPE_WIREGUARD`.

Here is a change which enables it for all secret types:

https://github.com/stv0g/gopacket/commit/82599fdade4a5248715f8989920ab76617cd3765